src/librustc_mir/shim.rs

```diff
@@ -120,6 +120,8 @@ fn make_shim<'tcx>(tcx: TyCtxt<'tcx>, instance: ty::InstanceDef<'tcx>) -> &'tcx
         &add_call_guards::CriticalCallEdges,
     ]);
 
+    result.ensure_predecessors();
```

Comment:
I didn't use the `ensure_predecessors_cache::EnsurePredecessorsCache` `MirPass` because the above pass was not always guaranteed to run. I came across one instance where a `MirPhase::Validated` pass had already been run on a `Body` by the time it arrived at this line, and so the above `run_passes` call did NOT run any of the passes. Adding the ensure pass at the `run_passes` for `MirPhase::Validated` would not have helped because the cache was being invalidated between that pass and this call. Having the single ensure outside of the `run_pass` list guaranteed that anyone using this body could call `Body::unwrap_predecessors()` without worrying about a panic.

Reply:
Leave this comment as a comment in the source